src/diagnostics/netcdf_writer.jl

```diff
@@ -497,13 +536,30 @@ function NetCDFWriter(;
         error("Cannot process hysography $hypsography")
     end
 
+    if disable_vertical_interpolation
+        # It is a little tricky to override the number of vertical points because we don't
+        # know if the vertical direction is the 2nd (as in a plane) or 3rd index (as in a
+        # box or sphere). To set this value, we check if we are on a plane or not
+
+        # TODO: Get the number of dimensions directly from the space
+        num_horiz_dimensions =
+            Spaces.horizontal_space(space) isa Spaces.SpectralElementSpace1D ?
+            1 : 2
```

Comment:
@simonbyrne is there a better way to do this?

Reply:
Not easily. We do define `coordinate_type` on some objects, but not grids or spaces.

Reply:
Okay, thanks!